.editorconfig

```diff
@@ -0,0 +1,26 @@
+root = true
+
+[*]
+charset = utf-8
+end_of_line = lf
+indent_size = 2
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+[Makefile]
+indent_size = 4
+indent_style = tab
+
+[*.robot]
+indent_size = 4
+indent_style = tab
+
+[*.{py,pyi,rst,toml}]
+indent_size = 4
+
+[*.md]
+trim_trailing_whitespace = false
+
+[*.{diff,patch}]
+trim_trailing_whitespace = false
```

Comment:
Good point:

For `md` files, there is a convention —that I do not agree with by the way— to insert new lines by leaving a trailing whitespace. So that rule is proposed to avoid breaking new lines inadvertently. However, I would not mind trimming whitespaces as a general rule, if that makes sense for everybody.

Regarding `diff` and `patch`, it comes from more or less the same reason above. Again, my preference is to trim whitespaces a a general rule if that suits everybody.

Reply:
I am clearly in favour of trimming whitespace. I believe Markdown forced line breaks should be implemented with a backslash rather than an invisible character at the end of the line.

For `diff` and `patch`, I totally see the reason for not trimming whitespace. I am just in favour of not adding rules for file types that are not checked into the repository and most likely never will: they add a maintenance burden for no reason.

Reply:
I have not approved because of that, but in any case, my review was a comment, nothing is blocking and you got an approval already, so feel free to merge as is if this removal suggestion is not fitting 🙂